### ISSUE TITLE
Feature Widget::setInputCallback() is incomplete.

### ISSUE THREAD

**Issue by discordier** [contao/core-bundle#1427]
The feature introduced in #674 is incomplete as not all widgets support them.

Some widgets override the `validate()`method and therefore do not handle the input callback at all.
These are (so far):
- CheckBox
- CheckBoxWizard
- FormCaptcha (don't know if we want to handle it, as it directly refers to `\Input`)
- FormCheckBox
- FormFileUpload (Pretty sure not able to implement easily with bc)
- FormRadioButton
- FormSelectMenu
- OptionWizard
- RadioButton
- SelectMenu

I wonder how to proceed here. Shall we rework each and every Widget or rather move the input callback handling into the `getPost()` method and fix it for all in one go?

**Comment by leofeyer** [comment]
@Toflar /cc

**Comment by leofeyer** [comment]
As discussed in Mumble on March 15th, we want to handle it in the `getPost()` method.

```diff
--- a/src/Resources/contao/library/Contao/Widget.php
+++ b/src/Resources/contao/library/Contao/Widget.php
@@ -776,8 +776,7 @@ abstract class Widget extends \Controller
 	 */
 	public function validate()
 	{
-		$varValue = (\is_callable($this->inputCallback) ? \call_user_func($this->inputCallback) : $this->getPost($this->strName));
-		$varValue = $this->validator($varValue);
+		$varValue = $this->validator($this->getPost($this->strName));
 
 		if ($this->hasErrors())
 		{
@@ -797,6 +796,11 @@ abstract class Widget extends \Controller
 	 */
 	protected function getPost($strKey)
 	{
+		if (\is_callable($this->inputCallback))
+		{
+			return \call_user_func($this->inputCallback);
+		}
+
 		if ($this->useRawRequestData === true)
 		{
 			/** @var Request $request */
```

**Comment by leofeyer** [comment]
Fixed in 1eeaf5431bdd0f4e16e74d9fd2309863080d1672.
